lib/rubyboy/bus.rb

```diff
@@ -11,74 +11,93 @@ def initialize(ppu, rom, ram, mbc, timer, interrupt, joypad, apu)
       @apu = apu
       @interrupt = interrupt
       @timer = timer
+    end
+
+    def read_byte(addr)
+      case addr >> 12
+      when 0x0, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0xa, 0xb
+        return @mbc.read_byte(addr)
+      when 0x8, 0x9
+        return @ppu.read_byte(addr)
+      when 0xc
+        return @ram.wram1[addr - 0xc000]
+      when 0xd
+        return @ram.wram2[addr - 0xd000]
+      when 0xf
+        case (addr >> 8) & (0xf)
```

Comment:
Would you mind changing it to be written as follows, without using an ampersand:

```
when 0xf
  case addr >> 8
  when 0xfe
    ~~
  when 0xff
    ~~
```